backend/main/src/Shared/Adapters/DB/NishikiTableClient.ts

```diff
@@ -193,26 +199,81 @@ export class NishikiDynamoDBClient {
 	 * @param groupId
 	 * @returns {GroupData | null} - the group data. If the group does not exist, it returns null.
 	 */
-	async getGroup(groupId: string): Promise<GroupData | null> {
-		const getGroupInput: GetItemInput = {
-			TableName: this.tableName,
-			Key: marshall({
-				PK: groupId,
-				SK: "Group",
-			}),
-		};
+	async getGroup(groupId: { groupId: string }): Promise<GroupData | null>;
+	/**
+	 * get a group by the container ID.
+	 * @param containerId
+	 * @returns {GroupData | null} - the group data. If the group does not exist, it returns null.
+	 * @throws {NishikiTableClientError} - if the container has more than one group.
+	 */
+	async getGroup(containerId: {
+		containerId: string;
+	}): Promise<GroupData | null>;
+	async getGroup(
+		id: { groupId: string } | { containerId: string },
+	): Promise<GroupData | null> {
+		// when the argument is the groupId.
+		if ("groupId" in id && id.groupId) {
+			const groupId = id.groupId;
+
+			const getGroupInput: GetItemInput = {
+				TableName: this.tableName,
+				Key: marshall({
+					PK: groupId,
+					SK: "Group",
+				}),
+			};
 
-		const command = new GetItemCommand(getGroupInput);
-		const response = await this.dynamoClient.send(command);
+			const command = new GetItemCommand(getGroupInput);
+			const response = await this.dynamoClient.send(command);
 
-		if (!response.Item) return null;
+			if (!response.Item) return null;
 
-		const unmarshalledData = unmarshall(response.Item);
+			const unmarshalledData = unmarshall(response.Item);
 
-		return {
-			groupId: unmarshalledData.PK,
-			groupName: unmarshalledData.GroupName,
-		};
+			return {
+				groupId: unmarshalledData.PK,
+				groupName: unmarshalledData.GroupName,
+			};
+		}
+		if ("containerId" in id && id.containerId) {
+			const containerId = id.containerId;
+
+			const queryGroupInput: QueryInput = {
+				TableName: this.tableName,
+				IndexName: GROUP_AND_CONTAINER_RELATIONSHIP,
+				KeyConditionExpression: "ContainerId = :containerId",
+				ExpressionAttributeValues: marshall({
+					":containerId": containerId,
+				}),
+			};
+
+			const command = new QueryCommand(queryGroupInput);
+			const response = await this.dynamoClient.send(command);
+
+			if (!(response.Items && response.Items.length > 0)) return null;
+
+			console.log(response.Items);
+
+			if (response.Items.length > 1) {
+				const containerIds: string[] = response.Items.map(
+					(item) => `ContainerId: ${unmarshall(item).ContainerId}`,
+				);
+
+				throw new NishikiTableClientError(
+					"There are more than one groups in the container.",
+					containerIds,
+				);
+			}
+
+			const result = unmarshall(response.Items[0]);
+
+			return this.getGroup({ groupId: result.PK });
```

Comment:
To fulfill the return type, this method calls `getGroup` method again. This looks redundant.

Reply:
@hitohata 
I understood your point. Sorry, that's my bad. I think we should change the table definition like updating the ProjectionType of the GroupContainerRelationship to All or INCLUDE.
But for now, how about keeping going with your implementation? Then, I'll make issues in the document repo and backend repo.

Reply:
@taniguchiiqqq 
OK, thank you. I'm looking forward to you opening new issues.